File: software/examples/delphi/ExampleThreshold.pas
program ExampleThreshold;

{$ifdef MSWINDOWS}{$apptype CONSOLE}{$endif}
{$ifdef FPC}{$mode OBJFPC}{$H+}{$endif}

uses
  SysUtils, IPConnection, BrickletAccelerometer;

type
  TExample = class
  private
    ipcon: TIPConnection;
    a: TBrickletAccelerometer;
  public
    procedure ReachedCB(sender: TBrickletAccelerometer; const x: smallint;
                        const y: smallint; const z: smallint);
    procedure Execute;
  end;

const
  HOST = 'localhost';
  PORT = 4223;
  UID = 'XYZ'; { Change to your UID }

var
  e: TExample;

{ Callback for acceleration threshold reached }
procedure TExample.ReachedCB(sender: TBrickletAccelerometer; const x: smallint;
                             const y: smallint; const z: smallint);
begin
    WriteLn(Format('Acceleration(X): %f g', [x/1000.0]));
    WriteLn(Format('Acceleration(Y): %f g', [y/1000.0]));
    WriteLn(Format('Acceleration(Z): %f g', [z/1000.0]));
    WriteLn('');
end;

procedure TExample.Execute;
begin
  { Create IP connection }
  ipcon := TIPConnection.Create;

  { Create device object }
  a := TBrickletAccelerometer.Create(UID, ipcon);

  { Connect to brickd }
  ipcon.Connect(HOST, PORT);
  { Don't use device before ipcon is connected }

  { Get threshold callbacks with a debounce time of 10 seconds (10000ms) }
  a.SetDebouncePeriod(10000);

  { Register threshold reached callback to procedure ReachedCB }
  a.OnAccelerationReached := {$ifdef FPC}@{$endif}ReachedCB;

  { Configure threshold for acceleration values X, Y or Z "greather than 2g" (unit is g/1000) }
  a.SetAccelerationCallbackThreshold('>', 2*1000, 0, 2*1000, 0, 2*1000, 0);

  WriteLn('Press key to exit');
  ReadLn;
  ipcon.Destroy; { Calls ipcon.Disconnect internally }
end;

begin
  e := TExample.Create;
  e.Execute;
  e.Destroy;
end.

File: software/examples/vbnet/ExampleThreshold.vb
Imports System
Imports Tinkerforge

Module ExampleThreshold
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Accelerometer Bricklet

    ' Callback subroutine for acceleration reached callback (parameters have unit g/1000)
    Sub AccelerationReachedCB(ByVal sender As BrickletAccelerometer, ByVal x As Short, _
                              ByVal y As Short, ByVal z As Short)
        Console.WriteLine("Acceleration[X]: " + (x/1000.0).ToString() + " g")
        Console.WriteLine("Acceleration[Y]: " + (y/1000.0).ToString() + " g")
        Console.WriteLine("Acceleration[Z]: " + (z/1000.0).ToString() + " g")
        Console.WriteLine("")
    End Sub

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim a As New BrickletAccelerometer(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Get threshold callbacks with a debounce time of 10 seconds (10000ms)
        a.SetDebouncePeriod(10000)

        ' Register acceleration reached callback to subroutine AccelerationReachedCB
        AddHandler a.AccelerationReached, AddressOf AccelerationReachedCB

        ' Configure threshold for acceleration "greater than 2 g, 2 g, 2 g" (unit is g/1000)
        a.SetAccelerationCallbackThreshold(">"C, 2*1000, 0, 2*1000, 0, 2*1000, 0)

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module
